File: src/main/java/gargoyle/ct/pref/impl/prop/CTPrefBytesProperty.java
package gargoyle.ct.pref.impl.prop;

import gargoyle.ct.pref.CTPreferencesProvider;
import org.jetbrains.annotations.NotNull;

public class CTPrefBytesProperty extends CTPrefProperty<byte[]> {
    public CTPrefBytesProperty(@NotNull CTPreferencesProvider provider, String name) {
        this(provider, name, new byte[0]);
    }

    public CTPrefBytesProperty(@NotNull CTPreferencesProvider provider, String name, byte[] def) {
        super(byte[].class, provider, name, def);
    }
}

File: src/main/kotlin/gargoyle/ct/pref/impl/prop/CTPrefBytesProperty.kt
package gargoyle.ct.pref.impl.prop

import gargoyle.ct.pref.CTPreferencesProvider

class CTPrefBytesProperty @JvmOverloads constructor(
    provider: CTPreferencesProvider,
    name: String,
    def: ByteArray = ByteArray(0)
) : CTPrefProperty<ByteArray>(
    ByteArray::class, provider, name, def
)
